Save Images to directory on batches only

nightred opened the pull request:
This will save images to a sub directory only when the job is larger then 1 image. Otherwise it will use the default image folder.

nightred committed:
Save Images to directory on batches only

This will save images to a sub directory only when the job is larger then 1 image.
Otherwise it will use the default image folder.

dfaker commented:
I think there's already a PR for this if this is addressing issue #1086 ?

#1090 does fixes the issue by setting the existing `grid` parameter when it's a grid.

nightred committed:
Update processing.py

pass that this is a grid for file save

nightred commented:
Fixed the 

> I think there's already a PR for this if this is addressing issue #1086 ?
> 
> #1090 does fixes the issue by setting the existing `grid` parameter when it's a grid.

This did not fix the issue with grids until i misunderstood what you said and added that fix.
What this does is make it so that single image batch's ignore the batch to subdirectory rule for the generated images not the grid.

nightred committed:
Update processing.py

ugly edit be gone please

AUTOMATIC1111 commented:
why would single image batches ignore the setting to write results  to t he directory

nightred commented:
This is likely my personal process but here is the idea.
I generate single images on a fixed seed to dial in the prompt, I do not want ever test image to have a separate folder. With all test images in a single folder I can compare them easily. 
Once the prompt is good I would run a large batch and i want all of that in a single folder.

This could be expanded to have an option for this function, I did not think on that yet.

nightred committed:
Update images.py

Check for setting to ignore subfolder on single image

nightred committed:
Update shared.py

Added option to ignore subdirectory when batch size is one

nightred commented:
Added an option to enable this feature

noprompt commented:
This just complicates the program. I don't think its worth the trouble.

noprompt commented:
For this particular case, I think would would make sense from a modularity and maintenance standpoint, would be to expose image saving as callback that a programmer could hook into. The callback could take the paths of the persisted images on disk and decide what to do with that on their own terms. There's a namespace which effectively holds the global state for the application, it seems like a good place to leave a place to leave an `on_images_saved` sized hole for your callback.

nightred commented:
Closing as not willing to make submodule
